libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -417,29 +405,27 @@ void YamlConfiguration::parseConnectionYaml(
 
 
         // Configure connection source
-        auto rawRelationship = connectionNode["source relationship name"]
-            .as<std::string>();
+        checkRequiredField(&connectionNode, "source relationship name", CONFIG_YAML_CONNECTIONS_KEY);
+        auto rawRelationship = connectionNode["source relationship name"].as<std::string>();
         core::Relationship relationship(rawRelationship, "");
-        logger_->log_debug(
-            "parseConnection: relationship => [%s]", rawRelationship);
+        logger_->log_debug("parseConnection: relationship => [%s]", rawRelationship);
         if (connection) {
           connection->setRelationship(relationship);
         }
 
         uuid_t srcUUID;
-        std::string connectionSrcProcName = connectionNode["source name"]
-            .as<std::string>();
+
         if (connectionNode["source id"]) {
-          std::string connectionSrcProcId = connectionNode["source id"]
-              .as<std::string>();
+          std::string connectionSrcProcId = connectionNode["source id"].as<std::string>();
 
           uuid_parse(connectionSrcProcId.c_str(), srcUUID);
         } else {
           // if we don't have a source id, try harder to resolve the source processor.
           // config schema v2 will make this unnecessary
+          checkRequiredField(&connectionNode, "source name", CONFIG_YAML_CONNECTIONS_KEY);
```

Comment:
If source id is present, yes. If id not present, there are two possible fallbacks that rely on the "source name" field. I used the Java implementation as a guide: 

From org.apache.nifi.minifi.commons.schema.v1.ConfigSchemaV1.java [1]
```
List<ConnectionSchema> connectionSchemas = new ArrayList<>(connections.size());
        for (ConnectionSchemaV1 connection : connections) {
            ConnectionSchema convert = connection.convert();
            convert.setId(getUniqueId(ids, convert.getName()));

            String sourceName = connection.getSourceName();
            if (remoteInputPortIds.contains(sourceName)) {
                convert.setSourceId(sourceName);
            } else {
                if (duplicateProcessorNames.contains(sourceName)) {
                    problematicDuplicateNames.add(sourceName);
                }
                String sourceId = processorNameToIdMap.get(sourceName);
                if (!StringUtil.isNullOrEmpty(sourceId)) {
                    convert.setSourceId(sourceId);
                }
            }

            String destinationName = connection.getDestinationName();
            if (remoteInputPortIds.contains(destinationName)) {
                convert.setDestinationId(destinationName);
            } else {
                if (duplicateProcessorNames.contains(destinationName)) {
                    problematicDuplicateNames.add(destinationName);
                }
                String destinationId = processorNameToIdMap.get(destinationName);
                if (!StringUtil.isNullOrEmpty(destinationId)) {
                    convert.setDestinationId(destinationId);
                }
            }
            connectionSchemas.add(convert);
        }
```

[1] https://github.com/apache/nifi-minifi/blob/master/minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/v1/ConfigSchemaV1.java